components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt

```diff
@@ -154,4 +167,46 @@ class EventsStorageEngineTest {
         assertEquals("[[0,\"telemetry\",\"test_event_clear\",\"test_event_object\",null,null]]",
             snapshot.toString())
     }
+
+    @Test
+    fun `test sending of event ping when it fills up`() {
+        val server = MockWebServer()
+        server.enqueue(MockResponse().setBody("OK"))
+
+        EventsStorageEngine.clearAllStores()
+        val click = EventMetricType(
+            disabled = false,
+            category = "ui",
+            lifetime = Lifetime.Ping,
+            name = "click",
+            sendInPings = listOf("default"),
+            objects = listOf("buttonA")
+        )
+
+        val realClient = Glean.httpPingUploader
+        val testConfig = Glean.configuration.copy(
+            serverEndpoint = "http://" + server.hostName + ":" + server.port
+        )
+        Glean.httpPingUploader = HttpPingUploader(testConfig)
+
+        try {
+            for (i in 0..499) {
+                click.record("buttonA")
```

Comment:
Can we encode `i` in some field of the event? To make sure we're actually sending different events. Maybe the `extra` field might be a good option.

Reply:
Good idea.